### PR TITLE
Fix more recursive failures

### PR DESCRIPTION
**Pull request by CUB3D**
This applies the same changes as in #67 to all other Value types that can have non-trivial children leading to reference cycles.

Also adds a fuzzer for the `parse_single_element` function. With this changeset neither `fuzz_amf0_body` `fuzz_amf3_body` or `fuzz_amf3_single_element` panic after ~500million fuzz cases each. 

Note this is not as clean as I would like, but I want to fix the panics first and worry about the refactor later

### CODE REVIEW

File: flash-lso/src/amf3/read.rs [hunk 1]
```diff
@@ -464,145 +473,225 @@ impl AMF3Decoder {
...
-                i,
-                Value::VectorObject(elems, object_type_name, fixed_length == 1),
-            ))
-        })
+                let id = if let Value::VectorObject(id, _, _, _) =
+                    this.object_reference_table.get(ofi).unwrap().as_ref()
+                {
+                    id
+                } else {
+                    unreachable!()
```
Comment (torokati44) [inline flash-lso/src/amf3/read.rs:561]: nit: A message would be appreciated here, for example like at `Value::ECMAArray`.

File: flash-lso/src/amf3/read.rs [hunk 2]
```diff
@@ -464,145 +473,225 @@ impl AMF3Decoder {
...
+                    .collect::<Vec<_>>();
 
-            Ok((i, Value::Dictionary(pairs, weak_keys == 1)))
-        })
+                let id = if let Value::Dictionary(id, _, _) =
+                    this.object_reference_table.get(ofi).unwrap().as_ref()
+                {
+                    id
+                } else {
+                    unreachable!()
```
Comment (torokati44) [inline flash-lso/src/amf3/read.rs:666]: nit: A message would be appreciated here, for example like at `Value::ECMAArray`.

File: flash-lso/src/amf3/read.rs (hunk 1 above)
Comment (CUB3D) [reply flash-lso/src/amf3/read.rs]: Done

File: flash-lso/src/amf3/read.rs (hunk 2 above)
Comment (CUB3D) [reply flash-lso/src/amf3/read.rs]: Done

**Pull request merged**